temporallib/worker/worker.py

```diff
@@ -29,12 +30,12 @@ class WorkerOptions:
 
 class Worker(TemporalWorker):
     """
-    A class which wraps the :class:`temporalio.client.Client` class
+    A class which wraps the :class:`temporalio.worker.Worker` class
     """
 
     def __init__(
         self,
-        client: Client,
+        client: Client | TemporalClient,
```

Comment:
I don't believe Python 3.8 supports `|`, you'd need to use `from typing import Union` and do `Union[Client, TemporalClient]`. Could you please test the lib locally with a sample workflow run to ensure it works as intended?

General note: we could do with an integration test for this lib.